### ISSUE TITLE
DenseFlipout accuracy and uncertainty issue

### ISSUE THREAD

**Issue by henrysky** [tensorflow/probability#166]
I actually have tried `DenseFlipout ` and `DenseReparameterization` but I am particularly interested in `DenseFlipout `. It seems like even in this simple y=mx+c regression  `DenseFlipout `  weights and bias is not accurate and uncertainty is unreasonably high. Is there anything I did wrong or there is something wrong in the implementation??

```
from tensorflow_probability.python.layers import DenseFlipout
import numpy as np
from tensorflow.python import keras
Input = keras.layers.Input
Model = keras.models.Model

# Data preparation
polynomial_coefficient = [0.1, -0.5]
random_xdata = np.random.normal(0, 1, (1000, 1))
random_ydata = polynomial_coefficient[1] * random_xdata + polynomial_coefficient[0]

input = Input(shape=[1, ])
output = DenseFlipout(units=1, name='polyfit')(input)
model = Model(inputs=input, outputs=output)
model.compile(optimizer='adam', loss='mse')

model.fit(random_xdata, random_ydata, batch_size=32, epochs=10)
print(model.get_layer("polyfit").get_weights())
```

**Comment by axch** [comment]
@dustinvtran Any thoughts on this?

**Comment by ImranRolo** [comment]
@henrysky @axch  You shouldn't be calculating the loss with mean squared error (alone). You need to use the full likelihood for regression.

**Comment by MFreidank** [comment]
@henrysky This has an example for linear regression: 
https://brendanhasz.github.io/2018/12/03/tfp-regression

Note: Using `DenseFlipout` seems to constrain one in the loss we can optimize, at least I have only seen it used with `ELBO` thus far.

**Comment by srvasude** [comment]
Closing this as I believe the issue has been answered. Feel free to reopen this if there are any more questions.
